modules/libraries/src/backend/utils.ts

```diff
@@ -0,0 +1,178 @@
+import * as sdk from 'botpress/sdk'
+import { spawn } from 'child_process'
+import fse from 'fs-extra'
+import path from 'path'
+
+import { isOffline, packageJsonPath, packageLockJsonPath, sharedLibsDir } from '.'
+
+const debug = DEBUG('libraries')
+
+const LIB_FOLDER = 'libraries/'
+
+const packageNameRegex = /^(@[a-z0-9-~][a-z0-9-._~]*\/)?[a-z0-9-~][a-z0-9-._~]*$/
+
+// Taken from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+const versionRegex = /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/
+
+const sanitizeArg = (text: string) => text.replace(/[^a-zA-Z0-9\/_.@^\-\(\) ]/g, '').replace(/\/\//, '/')
+
+export const validateNameVersion = ({ name, version }: { name: string; version?: string }) => {
+  const nameValid = name && packageNameRegex.test(name)
+  const versionValid = !version || (version && versionRegex.test(version))
+
+  console.log(nameValid, versionValid, name, version)
+  if (!nameValid || !versionValid) {
+    throw new Error('Invalid characters found')
+  }
+
+  return { name, version }
+}
+
+export const executeNpm = async (args: string[] = ['install'], customLibsDir?: string): Promise<string> => {
+  const moduleDir = process.LOADED_MODULES['libraries']
+  const nodeFolder = process.pkg ? 'node_production_modules' : 'node_modules'
+
+  if (isOffline) {
+    args.push('--offline')
+  }
+
+  // Hides superfluous messages
+  args.push('--no-fund')
+
+  // Necessary for post install scripts when running from the binary
+  args.push('--scripts-prepend-node-path')
+
+  const cleanArgs = args.map(x => sanitizeArg(x))
```

Comment:
This is all args manipulation & validation, can you extract that piece of code in a clearly named function  ?